### PR TITLE
Drop settings prop from HotTable

### PR DESCRIPTION
**Pull request by NOtherDev**
### Context

I don't see any usages of settings prop in docs or examples and it creates a dual interface - a user can pass things both directly into HotTable props or via settings object. Firstly, it's not clear which one wins in case of conflicts. Secondly, passing objects into props unnecessarily increase the risk of re-renders caused by new object instances being created on every parent render. Thirdly, it creates a bit of complexity to the internal code.

### How has this been tested?

Unit tests ensuring the preferred method of passing settings (in props directly) still works. But mostly code removal.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. Discussion in RFC: https://gist.github.com/evanSe/8c67315d837873c7f32f7446f027e5e7?permalink_comment_id=4824890#gistcomment-4824890

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [x] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by evanSe — APPROVED** [review]
A step forward to simplification of the react wrapper 😄

**Comment by NOtherDev** [comment]
Now @evanSe tell me please how should we proceed with PRs. I've opened this one against my fork and I can merge into my fork, and then we'll merge them all into the upstream repo in one global PR, is that a plan?
Recreating each of these PRs retrospectively in an upstream repo will be probably a PITA. Or maybe I should wait a few days and once the RFC is accepted (even silently), I could do PRs against the upstream directly? WDYT?

**Comment by evanSe** [comment]
>  and then we'll merge them all into the upstream repo in one global PR, is that a plan?

Recreating pull requests would be a significant inconvenience. The main issue with merging into the upstream develop branch at the moment is that we use that branch to release new versions. Therefore, we could either continue merging into a dedicated upstream branch or keep the changes here until we perform a final single merge into develop in the future.

I am in favor of whatever is most convenient for you.

**Comment by NOtherDev** [comment]
OK, merging it here then. Let's then keep this fork as this dedicated branch. If there's a feedback coming from the RFC, we'll handle it here as well.

**Comment by NOtherDev** [comment]
@evanSe I think I briefly mentioned it already, but let me double check with you before proceeding.
Given there's some development in the main repo around the React wrapper, and given Chris suggested me to work in the main repo already, I'd like to recreate the already-merged PRs on top of the main branch in the main repo. Do you agree? Should we push them through the normal review process, or handle separately, given they were already reviewed here?

**Comment by evanSe** [comment]
> @evanSe I think I briefly mentioned it already, but let me double check with you before proceeding. Given there's some development in the main repo around the React wrapper, and given Chris suggested me to work in the main repo already, I'd like to recreate the already-merged PRs on top of the main branch in the main repo. Do you agree? Should we push them through the normal review process, or handle separately, given they were already reviewed here?

I was going to suggest moving to the main repo, we should develop there in a separate branch and once we are happy we will merge it into develop (after a review from the rest of the team)
@NOtherDev

**Comment by NOtherDev** [comment]
> I was going to suggest moving to the main repo, we should develop there in a separate branch and once we are happy we will merge it into develop (after a review from the rest of the team) @NOtherDev

Then I think I need you to create that separate target branch in the main repo so that I can create PRs onto it.

**Comment by evanSe** [comment]
> > I was going to suggest moving to the main repo, we should develop there in a separate branch and once we are happy we will merge it into develop (after a review from the rest of the team) @NOtherDev
> 
> Then I think I need you to create that separate target branch in the main repo so that I can create PRs onto it.

@NOtherDev created https://github.com/handsontable/handsontable/tree/feature/improved-react

**Pull request merged**
